compiler/rustc_typeck/src/check/pat.rs

```diff
@@ -1029,17 +1030,59 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 // last sub-pattern. In the case of `A(x)` the first and last may coincide.
                 // This looks like:
                 //
-                // help: missing parenthesis
+                // help: missing parentheses
                 //   |
                 // L |     let A((x, y)) = A((1, 2));
                 //   |           ^    ^
                 [first, ..] => (first.span.shrink_to_lo(), subpats.last().unwrap().span),
             };
             err.multipart_suggestion(
-                "missing parenthesis",
+                "missing parentheses",
                 vec![(left, "(".to_string()), (right.shrink_to_hi(), ")".to_string())],
                 Applicability::MachineApplicable,
             );
+        } else if fields.len() > subpats.len() {
+            let after_fields_span = if pat_span == DUMMY_SP {
+                pat_span
+            } else {
+                pat_span.with_hi(pat_span.hi() - BytePos(1)).shrink_to_hi()
```

Comment:
Can you test whether this breaks in the case of a pattern like `Foo(a    ,    _    )` where `Foo` has 3 fields?

Reply:
Are the spaces inside the parentheses significant? Or do you mean `Foo(a, _)`?

Reply:
They are significant. I mean to test a case a case where there are multiple spaces between `(`, `a`, `,`, `_` and `)`:

```rust
Foo(   a   ,     _    )
```

Reply:
Added the test. It seems to work.